fractal-puppy/Engine
# Feature / Generalize UI Hover behaviour in Gameplay

This is an extension of the "No attack / Move" when pressing the menu buttons on the right side of the screen to all UI images / buttons.

---
- Added an option to ComponentButton and ComponentImage to Activate/Deactivate Hover Detection when Mouse 1 is pressed or Mouse 3 is pressed 
- The UIHovered  method of ModuleUI can be called to check if the UI is being hovered (you can specify if you want to retrieve behaviour for Mouse 1 and/or Mouse 3 )
- Player Attack checks if UI is being hovered for Mouse 1
- Player Movement checks if UI is being hovered for Mouse 3

---
TEST
- By default, when any mouse button is pressed while hovering a UI element, the player should not react (no attack or movement)
- Hover detection on UI Buttons and/or UI Images should be able to properly work. IE. if detection for Mouse 1 is disabled and Mouse 3 is enabled, the player should attack if Mouse1 is pressed over the UI element but not move if Mouse3 is pressed (I suggest to check it using the right side menu buttons since there are not multiple layers of ui elements stacked)
- Hover Detection over should be properly serialized (Save and Load)
